plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -622,6 +627,34 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
                 .build();
     }
 
+    @Override
+    public void validateScan(ConnectorSession session, ConnectorTableHandle handle)
+    {
+        IcebergTableHandle table = (IcebergTableHandle) handle;
+        Optional<PartitionSpec> partitionSpec = table.getPartitionSpecJson()
+                .map(partitionSpecJson -> PartitionSpecParser.fromJson(SchemaParser.fromJson(table.getTableSchemaJson()), partitionSpecJson));
+        if (partitionSpec.isEmpty() || partitionSpec.get().isUnpartitioned()) {
+            return;
+        }
+        if (isFilterOnPartitionTable(session)
+                && table.getEnforcedPredicate().equals(TupleDomain.all())) {
+            Set<IcebergColumnHandle> icebergConstraintColumns = table.getConstraintColumns().stream()
+                    .map(columnHandle -> (IcebergColumnHandle) columnHandle)
+                    .collect(Collectors.toSet());
+            Set<String> par1 = icebergConstraintColumns.stream().map(IcebergColumnHandle::getName).collect(toImmutableSet());
```

Comment:
Can you please rename par1 and par2 to something more descriptive? Could be something like `columnsWithPredicates` and `partitionColumns`, respectively